### ISSUE TITLE
Add `Array.join()` method

### ISSUE THREAD

**Issue by mageowl** [godotengine/godot-proposals#8105]
### Describe the project you are working on

Converting a list of strings to a hint string in `_get_property_list`

### Describe the problem or limitation you are having in your project

In Godot 3 we had `PoolStringArray.join()`, but now that has been removed with no alternative.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Join an array of strings with a seperator:
```gdscript
var array = ["Hello", "World"]
array.join(" ") # "Hello World"
```

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically the same as Array.join in JavaScript.
All items in the array are converted to strings, and then appended together with a separator.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be done with `reduce`, but its pretty clunky.
```gdscript
array.reduce(func (a, b):
    return a + " " + b
)
```

### Is there a reason why this should be core and not an add-on in the asset library?

Cannot extend default classes.

**Comment by dalexeev** [comment]
There is `String.join()` method:

```gdscript
print(", ".join(["1", "2", "3"])) # 1, 2, 3
```

**Comment by MikeFP** [comment]
Since that's basically a syntax sugar for `reduce`, I think a better solution is to make some common reduce operations available by default, then you could just:
```
array.reduce(concat) # joins the elements of a Array[String] in a single string
array.reduce(sum) # sums all elements of an Array[int]
```

And others. I found myself wanting a `sum` function for so long.
